libs/ballot-interpreter/src/hmpb-rust/timing_marks.rs

```diff
@@ -363,34 +393,148 @@ pub fn find_timing_mark_grid(
     Ok(timing_mark_grid)
 }
 
+/// Scores a potential timing mark in the image by examining the image to
+/// determine how many of the pixels in and around the rectangle are the
+/// expected luminosity.
+fn score_timing_mark_at_rect(ballot_image: &BallotImage, rect: &Rect) -> Option<f32> {
+    let image = &ballot_image.image;
+    let threshold = ballot_image.threshold;
+    let rect_height = rect.height() as i32;
+    let image_rect = Rect::new(0, 0, image.width(), image.height());
+    let rect_area = rect.width() * rect.height();
+
+    // look for black pixels in the rectangle
+    let rect_within_image = rect.intersect(&image_rect)?;
+    let rect_sub_image = GenericImageView::view(
+        image,
+        rect_within_image.left() as u32,
+        rect_within_image.top() as u32,
+        rect_within_image.width(),
+        rect_within_image.height(),
+    );
+    let black_pixel_count = rect_sub_image
+        .pixels()
+        .filter(|(_, _, luma)| luma.0[0] <= threshold)
+        .count();
+    let black_pixel_score = black_pixel_count as f32 / rect_area as f32;
+
+    let mut white_pixel_count = 0;
+    let mut possible_white_pixel_count = 0;
+
+    // crop the area to the left of the timing mark whose width is the same as
+    // the timing mark's height, then look for white pixels in the rectangle
```

Comment:
Why do we use the timing mark's height here to extend the _width_ to the left? Seems like an odd mix of dimensions that has me a tad confused. Something like `width/2` or `width/4` would be more what I would expect

Reply:
Oh, is it because we're looking above/below by `height` later?